smoke/LeafDevice/LeafDevice.cs

```diff
@@ -36,11 +36,11 @@ public async Task RunAsync()
             // This test assumes that there is an edge deployment running as transparent gateway.
             try
             {
-                await this.InitializeTrustedCertsAsync();
-                await this.GetOrCreateDeviceIdentityAsync();
-                await this.ConnectToEdgeAndSendDataAsync();
-                await this.VerifyDataOnIoTHubAsync();
-                await this.VerifyDirectMethodAsync();
+                await this.InitializeTrustedCertsAsync().ConfigureAwait(false);
```

Comment:
Did you talk to Varun about adding ConfigureAwait(false)?  Others look good to me.

Reply:
After discussing this, it was agreed to wait for the SDK to publish their fix rather than address the warnings using ConfigureAwait(false). This style is not used across our code base and we shouldn't have to do this here in this project. As a consequence I have had to disable warnings as errors for the time being. 

The only outstanding warning is "warning CA2007: Do not directly await a Task without calling ConfigureAwait", all others have been addressed.